publish/check-for-published-rpm.sh

```diff
@@ -0,0 +1,56 @@
+#!/usr/bin/bash
+set -exo pipefail
+
+# The architecture on which to expect a published RPM, e.g. el8.x86_64.
+arch=$1
+
+lsrpm () {
+    # Show published RPMs for $PACKAGE_NAME.
+    s3cmd ls "s3://alibuild-repo/RPMS/$arch/alisw-$PACKAGE_NAME+" |
+        # Trim leading modtime, file size, URL path and file extension, leaving
+        # only the package name, which we can pass to yum.
+        sed 's|^.*/||; s|\.rpm$||' | sort  # comm(1) expects lines in sorted order.
+}
+
+# Store list of RPMs for this package, so we can compare later.
+lsrpm > old-rpms.txt
+
+# Upload a canary file; this will be removed by the publishing script once it's
+# done, so we know when to check for our desired RPM. aliPublish has an instance
+# per architecture, so make per-architecture canary files.
+canary=$arch/$BUILD_TAG.finished
+date | s3cmd put - "s3://alibuild-repo/rpmstatus/$canary"
+while [ -n "$(s3cmd ls "s3://alibuild-repo/rpmstatus/$canary")" ]; do
+	sleep 60
+done
+
+# Now see if the RPM we want has been published.
+# Show only lines in "file 1", i.e. filenames that are only in S3.
+lsrpm | comm -23 - old-rpms.txt > new-rpms.txt
+
+# Check if there are any new RPMs.
+if [ -z "$(cat new-rpms.txt)" ]; then
+    echo "FAILED: aliPublish finished, but produced no new RPM for $PACKAGE_NAME" >&2
+    exit 1
+fi
+
+echo "SUCCESS: New RPMs for $PACKAGE_NAME follow:" >&2
```

Comment:
I would drop "SUCCESS" since you do not know if the package is actually installable at this point.

Reply:
Done!